sdk/anomalydetector/ai-anomaly-detector-rest/src/isUnexpected.ts

```diff
@@ -144,52 +130,36 @@ function geParametrizedPathSuccess(method: string, path: string): string[] {
     const candidatePath = getPathFromMapKey(key);
     // Get each part of the url path
     const candidateParts = candidatePath.split("/");
-
-    // If the candidate and actual paths don't match in size
-    // we move on to the next candidate path
-    if (
-      candidateParts.length === pathParts.length &&
-      hasParametrizedPath(key)
-    ) {
-      // track if we have found a match to return the values found.
-      let found = true;
-      for (let i = 0; i < candidateParts.length; i++) {
-        if (
-          candidateParts[i]?.startsWith("{") &&
-          candidateParts[i]?.endsWith("}")
-        ) {
-          // If the current part of the candidate is a "template" part
-          // it is a match with the actual path part on hand
-          // skip as the parameterized part can match anything
-          continue;
-        }
-
-        // If the candidate part is not a template and
-        // the parts don't match mark the candidate as not found
-        // to move on with the next candidate path.
-        if (candidateParts[i] !== pathParts[i]) {
-          found = false;
-          break;
-        }
+    // track if we have found a match to return the values found.
+    let found = true;
+    for (let i = candidateParts.length - 1; i >= 0; i--) {
+      if (candidateParts[i]?.startsWith("{") && candidateParts[i]?.endsWith("}")) {
+        // If the current part of the candidate is a "template" part
+        // it is a match with the actual path part on hand
+        // skip as the parameterized part can match anything
+        continue;
       }
 
-      // We finished evaluating the current candidate parts
-      // if all parts matched we return the success values form
-      // the path mapping.
-      if (found) {
-        return value;
+      // If the candidate part is not a template and
+      // the parts don't match mark the candidate as not found
+      // to move on with the next candidate path.
+      if (candidateParts[i] !== pathParts[i]) {
```

Comment:
pathParts.length not always equal with candidateParts.length. 

pathParts= [ '', 'anomalydetector', 'v1.1', 'multivariate', 'models' ]
candidateParts = [ '', 'multivariate', 'models' ]